modules/core/src/core/lib/deck.js

```diff
@@ -278,15 +288,34 @@ export default class Deck {
     });
   }
 
+  // Get the most relevant view state: props.viewState, if supplied, shadows internal viewState
+  // TODO: For backwards compatibility ensure numeric width and height is added to the viewState
   _getViewState(props) {
-    return Object.assign({}, props.viewState || {}, {
+    return Object.assign({}, props.viewState || this.viewState || {}, {
       width: this.width,
       height: this.height
     });
   }
 
   // Callbacks
 
+  _onViewStateChange({viewState}, ...args) {
+    // If initialViewState was set on creation, auto track position
+    if (this.viewState) {
+      this.viewState = viewState;
+      this.layerManager.setParameters({viewState});
+      this.controller.setProps({viewState});
+    }
+    // Let app know that view state has changed
+    if (this.props.onViewStateChange) {
+      this.props.onViewStateChange({viewState}, ...args);
+    }
+    // "HACK": Calling onResize to inform React component that it needs to update
+    if (this.props.onResize) {
```

Comment:
Alternatively this can be handled inside `DeckGL`. The React component can provide a `onViewStateChange` wrapper that calls the user callback and invokes a rerender.